generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -219,6 +219,16 @@ private Expr makeOpaqueZeroOrOne(boolean isZero, BasicType type, boolean constCo
           }
           return new FunctionCallExpr("sin", makeOpaqueZero(type, constContext, newDepth,
               fuzzer));
+        case 8:
+          // abs(opaque)
+          if (!BasicType.allGenTypes().contains(type)) {
+            final boolean isAbsIntSupported = shadingLanguageVersion.supportedAbsInt();
+            if (!isAbsIntSupported || !BasicType.allIGenTypes().contains(type)) {
+              continue; // abs operates only on gen and igen types.
```

Comment:
I would prefer you to split this up into separate checks:

if (!BasicTypes.allIGenTypes().contains(type)) {
  continue; // comment on readon
}
if ( ! the check for whether this type is supported ) {
  continue; // comment on the reason
}

In general for this project I favour lots of small checks for whether a transformation is applicable.